models/Asset.php

```diff
@@ -640,15 +644,6 @@ public function correctPath()
                 // use the parent's path from the database here (getCurrentFullPath), to ensure the path really exists and does not rely on the path
                 // that is currently in the parent asset (in memory), because this might have changed but wasn't not saved
                 $this->setPath(str_replace('//', '/', $parent->getCurrentFullPath() . '/'));
-            } else {
-                trigger_deprecation(
```

Comment:
I think we should also throw an exception if the parent with the given ID doesn't exist.

Reply:
You are right, added the exception + tests